### PR TITLE
[21081] AllocTest in 3.x

### PR DESCRIPTION
**Pull request by cferreiragonz**
## Description

This PR makes the AllocTests compatible with changes implemented in 3.x.

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- _N/A_ Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- _N/A_ Any new/modified methods have been properly documented using Doxygen. 
- _N/A_ Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- __NO__ Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- [X] Changes are API compatible. 
- _N/A_ New feature has been added to the `versions.md` file (if applicable).
- _N/A_ New feature has been documented/Current behavior is correctly described in the documentation. 
    
- _N/A_ Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- _N/A_ Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/profiling/allocations/AllocTestPublisher.hpp [hunk 1]
```diff
@@ -34,9 +34,9 @@
 #include <fastdds/dds/topic/Topic.hpp>
 #include <fastdds/dds/topic/TypeSupport.hpp>
 
-#include "AllocTestType.h"
+#include "AllocTestType.hpp"
 
-class AllocTestPublisher : public eprosima::fastdds::dds::DataWriterListener
```
Comment (MiguelCompany) [inline test/profiling/allocations/AllocTestPublisher.hpp:39]: This was like this on purpose, we'd like to remove the pattern of having a separate listener class, so when this test was created, it was the first to use the new approach.

Please revert this

File: test/profiling/allocations/AllocTestPublisher.cpp [hunk 2]
```diff
@@ -27,21 +27,20 @@
...
 #define CHECK_RETURN_CODE(ret) \
-    if (ReturnCode_t::RETCODE_OK != ret) \
+    if (RETCODE_OK != ret) \
     { \
         return false; \
     }
 
 #define CHECK_ENTITY_CREATION(entity) \
-    if (nullptr != entity) \
+    if (nullptr == entity) \
```
Comment (MiguelCompany) [inline test/profiling/allocations/AllocTestPublisher.cpp:43]: How could the old code be working?

File: test/profiling/allocations/AllocTestSubscriber.hpp [hunk 3]
```diff
@@ -34,9 +34,9 @@
 #include <fastdds/dds/topic/Topic.hpp>
 #include <fastdds/dds/topic/TypeSupport.hpp>
 
-#include "AllocTestType.h"
+#include "AllocTestType.hpp"
 
-class AllocTestSubscriber : public eprosima::fastdds::dds::DataReaderListener
```
Comment (MiguelCompany) [inline test/profiling/allocations/AllocTestSubscriber.hpp:39]: Same here

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: test/profiling/allocations/AllocTestPublisher.cpp (hunk 2 above)
Comment (cferreiragonz) [reply test/profiling/allocations/AllocTestPublisher.cpp:43]: No idea, I was receiving an instantaneous return after executing when I tried to run it.

**Review by MiguelCompany — APPROVED** [review]

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**